sdcm/provision/helpers/certificate.py

```diff
@@ -93,6 +98,18 @@ def install_encryption_at_rest_files(remoter):
     remoter.sudo("md5sum /etc/encrypt_conf/*.pem", ignore_status=True)
 
 
+def cached_ca(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if CA_CERT_FILE.exists() and CA_KEY_FILE.exists():
+            LOGGER.info(
+                "CA %s certificate and %s key already exist. Skipping creation.", CA_CERT_FILE, CA_KEY_FILE)
+            return
+        return func(*args, **kwargs)
+    return wrapper
+
+
+@cached_ca
```

Comment:
Why the built in @cache decorator isn't enough ?

Reply:
caching-like between tests I suppose - IIUC previous one from the same job are left.
I wonder why we need that wrappers and can't this code be moved to functions itself

Reply:
Correct, checking the file is to ensure the we do not try to recreate it in another test process